routes/marketdata.js

```diff
@@ -1,8 +1,58 @@
-var express = require('express');
-var router = express.Router();
+const express = require('express');
+const firebase = require('firebase');
+const config = require('./config.js');
+const router = express.Router();
 
-router.get('/', function(req, res, next) {
-	res.render('marketdata');
+if (!firebase.apps.length) {
+    firebase.initializeApp(config.config);
+}
+// Creates connection to database.
+const db = firebase.database();
+// Links to head of database.
+const ref = db.ref("live_weller");
+// Links to markets list.
+const marketsRef = ref.child("markets");
+
+router.get('/:marketName', function(req, res, next) {
+    const marketName = req.params.marketName;
+
+    marketsRef.once('value', function(snapshot) {
+        snapshot.forEach(function(childSnapshot) {
```

Comment:
This snippet looks like it’s pulling the entire database and iterating through all the market documents until it finds the correct one, which is pretty inefficient since it adds an order of magnitude to both performance and networking. We only need a single document, so we should use market ID as the route parameter instead of market name. Whenever a market is created, it should have a UUID (Universally Unique ID) generated for it, where the UUID is the market document’s key in the collection/snapshot, so we can specifically get that document instead of iterating through all the documents. This ties into having a proper schema though, so maybe we can leave this till that PR.

Reply:
Having a different PR for that would probably be good.  I figured out how to access the market info, but passing the name and address of the market that is currently used as the key through the url causes issues, especially when the address contains a pound, #, sign.

I know what to do when this comes up, and I'll get it done next.

Reply:
Yeah, I was thinking about possible solutions for that issue. Maybe just replace all the problem characters with an asterisk and do a regex check? Or hash the market name and address to a safe character list (e.g. convert the string to a hexadecimal one)?

Reply:
That's what I have right now to a degree, but a little inconsistency caused some issues.  I can likely make some adjustments here and there to make it work with instant access with the current key setup, although it would probably be better just to overhaul the whole thing and get the UUID method implemented.

Hashing the market name is a good idea, I'll make some notes on it tomorrow so I know which bits to fix up.